app/assess/models/file_factory.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import asdict
+from typing import NamedTuple
+
+from app.assess.helpers import generate_csv_of_application
+from app.assess.models.fund import Fund
+from app.assess.models.round import Round
+from app.pdf.full_application import FullApplicationPdfContext
+from app.pdf.full_application import generate_full_application_pdf
+from flask import abort
+from flask import send_file
+from fsd_utils import generate_text_of_application
+
+
+class ApplicationFileRepresentationArgs(NamedTuple):
+    fund: Fund
+    round: Round
+    short_id: str
+    form_name_to_title_map: dict
+    question_to_answer: dict
+    application_json: dict
+    all_uploaded_documents: list[dict]
+
+
+def _generate_text_of_application(args: ApplicationFileRepresentationArgs):
+    from app.assess.routes import download_file
```

Comment:
is this import here some python good practice for private functions. could we not move to top of file to reduce duplication?

Reply:
It's to avoid a circular import since `generate_file_content` is exported from this module and used on the routes, importing it locally to the function prevents the circular race condition when starting the app, in terms of convention, it's either this, or an absolute import - or a shared module with imports.  Not sure what to go with so just did this 🤷‍♂️

Reply:
hmm we could possibly move download_file/s() to a helpers file, that routes file is huge so might not be the worst idea. Happy for it to stay too, just a thought

Reply:
Good shout, have moved it in https://github.com/communitiesuk/funding-service-design-assessment/pull/383/commits/e64e45c9330e7cfbf1596faad79b94f23747bbdc